certbot/tests/compat/filesystem_test.py

```diff
@@ -0,0 +1,38 @@
+"""Unit test for security module."""
```

Comment:
In fact I added several significant tests already after the comment you mentionned. You can see them in https://github.com/certbot/certbot/blob/e3c725a348a04350261df0d803612d02a79e86db/certbot/tests/compat/security_test.py. Each function in `filesystem` is used in several significant meanings in these tests.

On top of that, all tests in certbot on windows are relying in a way or another to the correct behavior of `filesystem`.

Reply:
> On top of that, all tests in certbot on windows are relying in a way or another to the correct behavior of filesystem.

I disagree with this idea. Plenty of Certbot code calls these methods and they don't blow up which is great, but I think that in most cases when we call into our dependency's code (including the Python standard library), we're just assuming the code works. We don't go out of our way to test that it actually does what it's supposed to.

As an example, I made a trivial modification to this PR so that `everyone` always has full access in https://github.com/certbot/certbot/commit/6c1d7ad59aa406e633d2e763236885892b4a499f. Running our test suite on that PR, [everything](https://ci.appveyor.com/project/certbot/certbot/builds/24714983) [passes](https://travis-ci.com/certbot/certbot/builds/112691776) (besides `lint` which I was too lazy to fix).
 
I'm open to other ways of testing this code, but I think it is important that we do some kind of testing other than being happy with the fact it didn't raise an exception.

If we go the route of unit tests, what I'd personally like to see in this PR is:

1. A test that `chmod` properly handles symlinks on Windows.
2. A test that adding permissions for `everyone` properly adds an entry in the ACL.
3. Some tests that ACEs are created in the way we expect (e.g. system and admin have full access, read, write, and execute are translated in the way we expect for users and everyone, and that the group permission does not change the ACL).

Due to these tests being added, I'd personally make a new class in this file specifically for tests for `chmod`.

Again, I know this task isn't a lot of fun, but I think getting this code right is important and would like to ensure we have a systematic way to test that it works properly both now and in the future so we can prevent regressions.

Reply:
No, it was not fun. And you did to convince me that any low-level unit test here, as it largely reuses the same design that what it tests, would prevent a regression. I agree that what you made by modifying the everyone permissions could not be caught by current tests, but these new tests, if they fail, will not say that a business logic is broken, just that the implementation changed in some way. For me, it is more a bunch of code that we will need to maintain than an effective guard against a security breach.

A real test here would be to check at system level that no one except the user can effectively read a private key. This will be done later in certbot-ci.

Nonetheless, you have the unit tests you asked.